src/components/form/file-upload/file-upload.tsx

```diff
@@ -0,0 +1,177 @@
+import { Label } from '../../typography/label/label';
+import clsx from 'clsx';
+import { IconSize, IconUpload } from '../../../elements/icons';
+import { Button } from '../../button/button';
+import { ButtonSize } from '../../button/types';
+import { Variant } from '../../../utlis/types';
+
+export const FileUpload: FC<FileUploadProps> = ({
+    label = '',
+    labelFileSize = '',
+    labelButton = '',
+    Icon,
+    IconActionLeft,
+    IconActionRight,
+    onLoadFile,
+}) => {
+    const fileInputId = useId();
+    const inputReference = useRef<HTMLInputElement>(null);
+    const [dragIsOver, setDragIsOver] = useState(false);
+    const [isValidFileType, setIsValidFileType] = useState(true);
+    const [isValidFileSize, setIsValidFileSize] = useState(true);
+
+    const handleDragOver = (event: DragEvent<HTMLDivElement>) => {
+        event.preventDefault();
+        setDragIsOver(true);
+    };
+
+    const handleDragLeave = (event: DragEvent<HTMLDivElement>) => {
+        event.preventDefault();
+        setDragIsOver(false);
+    };
+
+    const handleDrop = (event: DragEvent<HTMLDivElement>) => {
+        event.preventDefault();
+        setDragIsOver(false);
+
+        // Fetch the file and check the validation
+        const droppedFile = event.dataTransfer.files[0];
+        droppedFile && checkFileType(droppedFile.type);
+        droppedFile && checkFileSize(droppedFile.size);
+
+        if (droppedFile && isValidFileType && isValidFileSize) {
+            onLoadFile && onLoadFile(droppedFile);
+        }
+
+        // Use FileReader to read file content
+        const reader = new FileReader();
+
+        reader.onloadend = () => {
+            console.log(reader.result);
+        };
+
+        reader.onerror = () => {
+            console.error('There was an issue reading the file.');
+        };
+
+        reader.readAsDataURL(droppedFile);
+        console.log('reader', reader);
+    };
+
+    // Fire the input when the Button is clicked
+    const onFileUpload = () => {
+        inputReference.current?.click();
+    };
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+
+        event.target.files && checkFileType(event.target.files[0].type);
+        event.target.files && checkFileSize(event.target.files[0].size);
+
+        if (event.target.files && event.target.files[0] && isValidFileType && isValidFileSize) {
+            onLoadFile && onLoadFile(event.target.files[0]);
+        }
+    };
+
+    const checkFileType = (type: string): void => {
+        if (type === 'image/png') {
+            setIsValidFileType(true);
+        } else if (type === 'image/jpeg') {
+            setIsValidFileType(true);
+        } else {
+            setIsValidFileType(false);
+        }
+    };
+
+    const checkFileSize = (size: number): void => {
+        // File Size must be less than 5 MB
+        return size <= 52428800 ? setIsValidFileSize(true) : setIsValidFileSize(false);
```

Comment:
Better set this via props instead of hardcoded number, so the size is exchangeable if needed